html/semantics/forms/the-label-element/labelable-elements.html

```diff
@@ -91,8 +91,23 @@
 }, "Check if the hidden input element is not a labelable element.");
 
 test(function() {
-  testLabelsAttr("testHidden", null);
-}, "Check if the hidden input element can access 'labels'");
+  var hiddenInput = document.getElementById("testHidden");
+  assert_equals(hiddenInput.labels, null, "input[type=hidden] must have null .labels");
+
+  this.add_cleanup(function () {
+    hiddenInput.type = "hidden";
+  });
+
+  hiddenInput.type = "text";
+  testLabelsAttr("testHidden", "lbl5");
+  var labels = hiddenInput.labels;
+
+  hiddenInput.type = "hidden";
+  assert_equals(labels.length, 0, "Retained .labels NodeList should be empty after input type changed to hidden");
+
+  hiddenInput.type = "text";
```

Comment:
Can you change this to `checkbox` and add an assertion for `labels.length` after (should be 1 but I seem to get 0 in chromium and webkit) ?